### PR TITLE
cats Validated/ValidatedNel types integration

### PR DESCRIPTION
**Pull request by krzemin**
Integration is done as a separate module. Need to update documentation about this integration yet.

### CODE REVIEW

File: octopusCats/src/main/scala/octopus/cats/package.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package octopus
+
+import _root_.cats.data.{Validated, ValidatedNel, NonEmptyList}
```
Comment (MateuszKubuszok) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: Automatic import from IntelliJ?

File: octopusCats/src/main/scala/octopus/cats/package.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
```
Comment (MateuszKubuszok) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: I would prefer to `import cats.syntax.validated._` and then `value.valid` or `value.invalid`

File: octopusCats/src/main/scala/octopus/cats/package.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+package object cats {
+
+  implicit class OctopusCatsOps[T](val vr: ValidationResult[T]) extends AnyVal {
+
+    def toValidatedNel: ValidatedNel[ValidationError, T] = vr.errors match {
+      case Nil =>
+        Validated.Valid(vr.value)
+      case head :: tail =>
+        Validated.Invalid(NonEmptyList.of(head, tail : _*))
```
Comment (MateuszKubuszok) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: `NonEmptyList.fromList(vr.errors).map(_.invalid).orElse(vr.value.valid)`?

File: octopusCats/src/main/scala/octopus/cats/package.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    def toValidated: Validated[List[ValidationError], T] = vr.errors match {
```
Comment (MateuszKubuszok) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: I think if-else would be enough

File: octopusCats/src/main/scala/octopus/cats/package.scala (hunk 1 above)
Comment (krzemin) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: Nope. Package has the same name, root import is needed for the compiler to see the correct path.

File: octopusCats/src/main/scala/octopus/cats/package.scala (hunk 4 above)
Comment (krzemin) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: Tried that, could work as well, but with pattern match I don't have to transform errors in any way and can construct NEL instance directly from extracted `head` & `tail`.

File: octopusCats/src/main/scala/octopus/cats/package.scala (hunk 3 above)
Comment (krzemin) [inline octopusCats/src/main/scala/octopus/cats/package.scala]: Can be as well, both are equivalent. But for me current version seems to be shorter and more readable.

**Pull request merged**